update iceberg procedures according to #22608 (#22787)
prestodb/presto: presto-docs/src/main/sphinx/connector/iceberg.rst

     INSERT    |       2 |  677209275408372885 | {orderkey=18016, custkey=403, orderstatus=O, totalprice=174070.99, orderdate=1996-03-19, orderpriority=1-URGENT, clerk=Clerk#000000629, shippriority=0, comment=ly. quickly ironic excuses are furiously. carefully ironic pack}
     INSERT    |       2 |  677209275408372885 | {orderkey=18017, custkey=958, orderstatus=F, totalprice=203091.02, orderdate=1993-03-26, orderpriority=1-URGENT, clerk=Clerk#000000830, shippriority=0, comment=sleep quickly bold requests. slyly pending pinto beans haggle in pla}

Procedures
----------

> Use the CALL statement to perform data manipulation or administrative tasks.Procedures are available in the system schema of each catalog iceberg. The following are several examples:

Reviewer: ```suggestion
Use the CALL statement to perform data manipulation or administrative tasks. Procedures are available in the ``system`` schema of the catalog. 
```

Author: OK

> Register table

Author: OK

^^^^^^^^^^^^^^

Iceberg tables for which table data and metadata already exist in the

> file system can be registered with the catalog using the ``register_table``

Reviewer: ```suggestion
file system can be registered with the catalog. Use the ``register_table``
```
Suggest breaking up this single sentence paragraph to improve readability.

> procedure on the catalog's ``system`` schema by supplying the target schema,

Reviewer: ```suggestion
procedure on the catalog's ``system`` schema and supply the target schema,
```
Minor change to support new sentence creation in the previous line.

desired table name, and the location of the table metadata::

    CALL iceberg.system.register_table('schema_name', 'table_name', 'hdfs://localhost:9000/path/to/iceberg/table/metadata/dir')

.. note::

    If multiple metadata files of the same version exist at the specified
    location, the most recently modified one will be used.

A metadata file can optionally be included as an argument to ``register_table``
in the case where a specific metadata file contains the targeted table state::

    CALL iceberg.system.register_table('schema_name', 'table_name', 'hdfs://localhost:9000/path/to/iceberg/table/metadata/dir', '00000-35a08aed-f4b0-4010-95d2-9d73ef4be01c.metadata.json')

.. note::

    When registering a table with the Hive metastore, the user calling the

>     procedure will be set as the owner of the table and will have ``SELECT``,

Reviewer: ```suggestion
    procedure is set as the owner of the table and will have ``SELECT``,
```

    ``INSERT``, ``UPDATE``, and ``DELETE`` privileges for that table. These
    privileges can be altered using the ``GRANT`` and ``REVOKE`` commands.

.. note::

    When using the Hive catalog, attempts to read registered Iceberg tables
    using the Hive connector will fail.

> Unregister table

Author: OK

^^^^^^^^^^^^^^^^

Iceberg tables can be unregistered from the catalog using the ``unregister_table``
procedure on the catalog's ``system`` schema::

    CALL iceberg.system.unregister_table('schema_name', 'table_name')

.. note::

    Table data and metadata will remain in the filesystem after a call to
    ``unregister_table`` only when using the Hive catalog. This is similar to
    the behavior listed above for the ``DROP TABLE`` command.

> Rollback to Snapshots

Reviewer 2: ```suggestion
Rollback to Snapshot
```

Sorry to be pedantic - I think it should match the procedure name

Reviewer: When this is edited, also edit the next line "^^^^^^^^^^^^^^^^^^^^^" to match the length of the characters in the text of the heading

> Rollback to snapshots

Author: OK

^^^^^^^^^^^^^^^^^^^^^

> Roll back a table to a specific snapshot ID. Iceberg can roll back to a specific snapshot ID by using ``rollback_to_snapshot`` procedure on iceberg`s ``system`` schema::

Author: OK

    CALL iceberg.system.rollback_to_snapshot('table_name', 'snapshot_id');

The following arguments are available:

===================== ========== =============== =======================================================================
Argument Name         required   type            Description
===================== ========== =============== =======================================================================
``table``             ✔️          string          Name of the table to update

``snapshot_id``       ✔️          long            Snapshot ID to rollback to
===================== ========== =============== =======================================================================

> Expire snapshots

Author: ok
